File: py/2900_Longest_Unequal_Adjacent_Groups_Subsequence_I.py
class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        ret = []
        cur_val = groups[0]
        ret.append(words[0])

        for i in range (1, len(groups)):
            if (not cur_val and groups[i]) or (cur_val and not groups[i]):
                ret.append(words[i])
                cur_val = groups[i]

        return ret

File: cpp/2900_Longest_Unequal_Adjacent_Groups_Subsequence_I.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<string> getLongestSubsequence(vector<string>& words, vector<int>& groups) {
		vector<string> ret;
		int cur_val = groups[0];
		ret.push_back(words[0]);
		for (int i = 1; i < groups.size(); ++i) {
			if (!cur_val && groups[i] || 
                cur_val && !groups[i]) {
				ret.push_back(words[i]);
				cur_val = groups[i];
			}
		}

		return ret;
    }
};

